src/storage/v2/schema_info.hpp

```diff
@@ -0,0 +1,756 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include <algorithm>
+#include <boost/container_hash/hash_fwd.hpp>
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <mutex>
+#include <shared_mutex>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+
+#include "storage/v2/delta.hpp"
+#include "storage/v2/edge.hpp"
+#include "storage/v2/edge_accessor.hpp"
+#include "storage/v2/edge_ref.hpp"
+#include "storage/v2/enum_store.hpp"
+#include "storage/v2/id_types.hpp"
+#include "storage/v2/name_id_mapper.hpp"
+#include "storage/v2/property_value.hpp"
+#include "storage/v2/transaction.hpp"
+#include "storage/v2/vertex.hpp"
+#include "utils/logging.hpp"
+#include "utils/rw_spin_lock.hpp"
+#include "utils/small_vector.hpp"
+
+// TODO Add namespace schema_info
+namespace memgraph::storage {
+
+utils::small_vector<LabelId> GetCommittedLabels(const Vertex &vertex);
+
+utils::small_vector<LabelId> GetLabels(const Vertex &vertex, uint64_t commit_timestamp);
+
+struct PropertyTypeHash {
+  size_t operator()(const ExtendedPropertyType &type) const {
+    size_t seed = 0;
+    boost::hash_combine(seed, type.type);
+    boost::hash_combine(seed, type.temporal_type);
+    boost::hash_combine(seed, type.enum_type.value_of());
+    return seed;
+  }
+};
+struct PropertyInfo {
+  int n{0};
+  std::unordered_map<ExtendedPropertyType, int, PropertyTypeHash> types;
+
+  nlohmann::json ToJson(const EnumStore &enum_store, std::string_view key, uint32_t max_count) const;
+};
+struct TrackingInfo {
+  int n{0};
+  std::unordered_map<PropertyId, PropertyInfo> properties;
+
+  nlohmann::json ToJson(NameIdMapper &name_id_mapper, const EnumStore &enum_store) const;
+};
+
+using VertexKey = utils::small_vector<LabelId>;
```

Comment:
How do we ensure the Labels are sorted? (so we have stable hash)

Reply:
The hash itself is order independent.
That's also why I don't use hash_combine.

Reply:
Put a comment in for that please.